src/parallel.cpp

```diff
@@ -585,6 +585,7 @@ void HandleBlockMessageThread(CNode *pfrom, const string strCommand, CBlockRef p
 
         // the current fork is bad due to this block so reset the best header to the best fully-validated block
         // so we can download another fork of headers (and blocks).
+        LOCK(cs_main);
         CBlockIndex *mostWork = FindMostWorkChain();
```

Comment:
cs_main should already be locked here, not sure why clang would complain, maybe because of the boost scope guard again, but doesn't hurt to lock it again for a short while.

Reply:
It can't be  because "boost::thread thread(boost::bind(&HandleBlockMessageThread, pfrom, strCommand, pblock, inv));"